App/Services/EventHandlers/TextileNodeEventHandler.ts

```diff
@@ -10,112 +9,108 @@ import { contactsActions } from '../../features/contacts'
-    })
+        if (type_url === '/Join' ||
+            type_url === '/Leave') {
+          // Every time the a JOIN or LEAVE block is detected, we should refresh our in-mem contact list
+          // Enhancement: compare the joiner id with known ids and skip the refresh if known.
+          this.store.dispatch(contactsActions.getContactsRequest())
+        }
 
-    // TextileEventsActions
-    this.events.addListener('newNodeState', (payload) => {
-      this.store.dispatch(TextileEventsActions.newNodeState(payload.state))
-    })
-    this.events.addListener('startNodeFinished', () => {
-      this.store.dispatch(TextileEventsActions.startNodeFinished())
-    })
-    this.events.addListener('stopNodeAfterDelayStarting', () => {
-      this.store.dispatch(TextileEventsActions.stopNodeAfterDelayStarting())
-    })
-    this.events.addListener('stopNodeAfterDelayCancelled', () => {
-      this.store.dispatch(TextileEventsActions.stopNodeAfterDelayCancelled())
-    })
-    this.events.addListener('stopNodeAfterDelayFinishing', () => {
-      this.store.dispatch(TextileEventsActions.stopNodeAfterDelayFinishing())
-    })
-    this.events.addListener('stopNodeAfterDelayComplete', () => {
-      this.store.dispatch(TextileEventsActions.stopNodeAfterDelayComplete())
-    })
-    this.events.addListener('appStateChange', (payload) => {
-      this.store.dispatch(TextileEventsActions.appStateChange(payload.previousState, payload.newState))
-    })
-    this.events.addListener('updateProfile', () => {
-      this.store.dispatch(TextileEventsActions.updateProfile())
-    })
-    this.events.addListener('error', (payload) => {
-      this.store.dispatch(TextileEventsActions.newErrorMessage(payload.type, payload.message))
-    })
-    // Account actions
-    this.events.addListener('setRecoveryPhrase', (payload) => {
-      this.store.dispatch(AccountActions.setRecoveryPhrase(payload.recoveryPhrase))
-    })
-    this.events.addListener('walletInitSuccess', () => {
-      this.store.dispatch(AccountActions.initSuccess())
-    })
+        // create a local log line for the threadUpdate event
+        const message = `BlockType ${type_url} on ${update.thread}`
+        this.store.dispatch(DeviceLogsActions.logNewEvent((new Date()).getTime(), 'onThreadUpdate', message, false))
+      })
+    )
+    this.subscriptions.push(
+      Textile.events.addThreadAddedListener((threadId) => {
+        this.store.dispatch(PhotoViewingActions.threadAddedNotification(threadId))
+      })
+    )
+    this.subscriptions.push(
+      Textile.events.addThreadRemovedListener((threadId) => {
+        this.store.dispatch(PhotoViewingActions.threadRemoved(threadId))
+      })
+    )
+    this.subscriptions.push(
+      Textile.events.addNotificationReceivedListener((notification) => {
+        this.store.dispatch(NotificationActions.newNotificationRequest(toTypedNotification(notification)))
```

Comment:
The new event subscription API provides callbacks with typed data (`notification` here is a `INotification` type)